### PR TITLE
Removing Fabric component dependency

### LINKED ISSUE #6528: Always use fonts defined in the theme for components (remove Fabric component dependency)
Expected:

render any Fabric component, renders the correct fonts.

Resulted:

times new roman

### PR DESCRIPTION
**Pull request by aneeshack4**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #6528
- [x] Include a change request file using `$ npm run change`

#### Description of changes

In #6647, I loaded fonts theme for all components that have a theme available. In this PR, I'm removing the Fabric component dependency altogether by getting rid of the <Fabric> wrapper in FabricDecorator. This caused a lot of screener failures that exposed where font was not defined for a lot of components. I went through each component and if it's scss-based (has a .scss file) then I added a <Fabric> wrapper to that component's references in its respective stories.tsx files because we want to avoid changing these. For the non-scss based components, I believe we need to accept the new changes.

Components that **don't have a .scss file** (so I **didn't add a Fabric wrapper** to these components in their stories.tsx file): Callout, ColorPicker, Fonts, GroupedList, HoverCard, Icon, Link, List, Pivot, Spinner, TagPicker, TextField, Tooltop.

Components that **have a .scss file** (so I **added a Fabric wrapper** to these components in their stories.tsx file): DateField, DocumentCard, FolderCover, OverflowSet, PeoplePicker, ScrollablePane, Signals, SpinButton.

(note: initially started this work with #6926 but had to stop working on that during Shield and when I tried to use that branch again, there was a lot of weirdness with build errors so I started fresh hence the new PR)

#### Focus areas to test

The screener changes should only be on the non-scss based components. See if they can be accepted as new.

### CODE REVIEW

File: apps/vr-tests/src/stories/HoverCard.stories.tsx [hunk 1]
```diff
@@ -20,20 +16,17 @@ const expandingCardProps = {
...
-  ), { rtl: true });
\ No newline at end of file
+  .addDecorator(story => <Screener steps={new Screener.Steps().snapshot('default', { cropTo: '.testWrapper' }).end()}>{story()}</Screener>)
+  .addStory(
+    'Root',
+    () => (
+      <HoverCard
+        expandingCardProps={expandingCardProps}
+        instantOpenOnClick={true}
+        //styles={{ root: { fontFamily: 'Segoe UI', color: rgb2hex(51, 51, 51) } }}
```
Comment (natalieethell) [inline apps/vr-tests/src/stories/HoverCard.stories.tsx]: nit: cleanup :)

**Comment by aneeshack4** [comment]
Note: I know I can run "npm run screener:local" to test my changes locally & submit them all at once but the local screener results are not consistent with the results that come from this PR - I don't know why this is - for example, screener on this PR branch shows that some of the rejected changes have been fixed (or in the changes bucket once again to be reconsidered for approval): 
image
But the local test's results are a whole other story & show that all states have changed which is obviously incorrect because these two were run off at same code around the same time....
image

**Comment by aneeshack4** [comment]
@dzearing I think I've addressed all of the failures that were rejected... Fonts, I believe we discussed in person and you said the changes are fine so I think you might have to accept those? Also, apologies for the large # of commits and I just realized the last two are not named well. As I mentioned in the previous comment, Screener has been a struggle to work with in fixing all these visual regressions and has caused me to commit a lot of times changes to the same components' tests in order to test out fixes.

**Comment by aneeshack4** [comment]
@betrue-final-final I fixed the ColorPicker failures that were rejected. And fixed the FolderCover ones mostly too except for the font-family because I haven't been able to find a way to set fontFamily as a style in the FolderCover.stories.tsx file. Same with Fonts - when I set color="#333333" on each <p> in Fonts.stories.tsx, it doesn't change anything about the color but @dzearing and I spoke about Fonts in person and if I remember correctly, these color changes in Fonts can be approved. If y'all have any ideas of what I can do for FolderCover's fontfamily, please let me know!

**Comment by aneeshack4** [comment]
Some notes on my investigation into fixing the last few changes.

**FolderCover's fontfamily**: there's no "IStyle" whatsoever in all of the FolderCover code so setting fontFamily in the stories file is not an option and the only place anything related to font is being set is FolderCover.scss under metadata. I tried, among other things, to do "font-family: 'Segoe UI';" but the font family of the screener stories stayed Times New Roman. When I try to wrap FolderCover component in FolderCover.stories.tsx, it breaks most of the component and deletes the img: 
image

**Icon's fontSize**: first of all, clicking on the this screener result freezes the whole screen before you can open the left side panel to see what's wrong. So we have to guess from the picture - it looks like fontSize increased so I added styles={{root: { fontSize: '14px' } }} and tried a couple different sizes but none of them changed the size or resolved this change. 
image

**Fonts' color**: I tried to add color=rgb2hex(51, 51, 51) and color="#333333" and even color="red" to the   <div style={props.style}> wrapping the list of Ps and I tried to add a color individually to each of the Ps and I tried to add a color in the props of RepresentativeText in const RepresentativeText = (props: { style: React.CSSProperties }) but none of these work.
image

I've exhausted my list of ideas I can try so I think these changes either need to be accepted or if anyone has more suggestions on things I can try, I will do that!

File: apps/vr-tests/src/stories/DocumentCard.stories.tsx [hunk 2]
```diff
@@ -74,64 +75,80 @@ let previewPropsCompact = {
...
-    <DocumentCard onClickHref="http://bing.com">
-      <DocumentCardPreview {...previewProps} />
-      <DocumentCardTitle
-        title="Large_file_name_with_underscores_used_to_separate_all_of_the_words_and_there_are_so_many_words_it_needs_truncating.pptx"
-        shouldTruncate={false}
-      />
-      {DocActivity}
-    </DocumentCard>
+    <Fabric>
+      {' '}
```
[the remark this answers is not in the document]
Comment (aneeshack4) [reply apps/vr-tests/src/stories/DocumentCard.stories.tsx]: ooh no I'll clean this up - they just get randomly added & I thought I got all of them so thanks for catching these!

File: apps/vr-tests/src/stories/DocumentCard.stories.tsx [hunk 3]
```diff
@@ -74,64 +75,80 @@ let previewPropsCompact = {
...
-  <DocumentCardActivity
-    activity="Created a few minutes ago"
-    people={[{ name: 'Annie Lindqvist', profileImageSrc: TestImages.personaFemale }]}
-  />
+  <Fabric>
+    {' '}
+    <DocumentCardActivity
+      activity="Created a few minutes ago"
+      people={[{ name: 'Annie Lindqvist', profileImageSrc: TestImages.personaFemale }]}
+    />{' '}
```
Comment (khmakoto) [inline apps/vr-tests/src/stories/DocumentCard.stories.tsx]: Here too

File: apps/vr-tests/src/stories/DocumentCard.stories.tsx [hunk 4]
```diff
@@ -74,64 +75,80 @@ let previewPropsCompact = {
...
-    </DocumentCard>
+    <Fabric>
+      <DocumentCard onClickHref="http://bing.com">
+        <DocumentCardPreview {...previewProps} />
+        <DocumentCardTitle
+          title="Large_file_name_with_underscores_used_to_separate_all_of_the_words_and_there_are_so_many_words_it_needs_truncating.pptx"
+          shouldTruncate={true}
+        />
+        {DocActivity}
+      </DocumentCard>{' '}
```
Comment (khmakoto) [inline apps/vr-tests/src/stories/DocumentCard.stories.tsx]: And here

**Pull request merged**